src/samples/flow/compass/Makefile

```diff
@@ -1,7 +1,2 @@
 sample-$(FLOW_COMPASS_LSM303_SAMPLE) += compass-lsm303
 sample-compass-lsm303-$(FLOW_COMPASS_LSM303_SAMPLE) := lsm303.fbp
-sample-compass-lsm303-$(FLOW_COMPASS_LSM303_SAMPLE)-deps := \
-	flow/accelerometer.mod \
-	flow/compass.mod \
-	flow/console.mod \
-	flow/magnetometer.mod
```

Comment:
On Fri, 18 Sep 2015 15:08:53 -0700, Gustavo Sverzut Barbieri wrote:
> > @@ -1,7 +1,2 @@
> >  sample-$(FLOW_COMPASS_LSM303_SAMPLE) += compass-lsm303
> >  sample-compass-lsm303-$(FLOW_COMPASS_LSM303_SAMPLE) := lsm303.fbp
> > -sample-compass-lsm303-$(FLOW_COMPASS_LSM303_SAMPLE)-deps := \
> > -	flow/accelerometer.mod \
> > -	flow/compass.mod \
> > -	flow/console.mod \
> > -	flow/magnetometer.mod
> 
> I don't think so, you don't need them to compile your code -- I think. Linking will happen at a later phase.
> 
> Try to make the sample/module before the module (explicit target), see if dependencies are pulled in.

We don't need them linking, but we do need to fix up the build to make
the dependency clear. Right now I can build the samples and the modules
don't get built.